frameworks/js-bindings/bindings/manual/network/XMLHTTPRequest.cpp

```diff
@@ -724,10 +731,28 @@ JS_BINDED_FUNC_IMPL(MinXmlHttpRequest, send)
     _setHttpRequestHeader();
     _sendRequest(cx);
     _notify(_onloadstartCallback);
+    
+    //begin schedule for timeout
+    if(_timeout > 0)
+    {
+        _scheduler->scheduleUpdate(this, 0, false);
```

Comment:
1. are you sure the xhr timeout means timeout for connect in the w3c specification?
2. when you notify the callback, it's the right time when time's out? not some time delayed?